---
title: "Untitled"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
#General
library(tidyverse)
library(stringr)
library(ggplot2)
library(dplyr)
library(readr)
library(vegan)

#Clustering
library(recluster)
library(phytools)
library(maps)
library(stats)
library(cluster)
library(ape)
library(ggtree)
library(treeio)
library(tidytree)
library(ggstance)
library(ggnewscale)
library(grid)
library(gtable)

#Final figure
library(patchwork)
library(ggplotify)
library(cowplot)


```

Aim:

Combined figure for enriched GO terms based on Upsetplot of changes in gene expression 30 min after settlement induction.
Induced data are split into one plot for upregulation and one plot for downregulation. Each plot consists of 3 parts: 
1. change only in aboral induced
2. changes in both larval parts
3.changes only oral
Next to plot is an elongated larvae with aboral facing down next to phylogenetic GO terms with barplot numbers and manual annotation coding
See 
D:/Dropbox/PhD JCU/Chapter2_Gene expression/A.mil_oral_aboral_rnaseq/04_Functional_analysis/04_4_TopGO/TopGO_Output/02_Upset_up_down/TopGO_figure/Figure2_vision.pdf


For figures see 10_Upset_ORA_2021_04_19.Rmd
Also check out 11_Upset_boxplot.Rmd


Figure 2
2. What gets up and downregulated after settlement induction?
  With facet wrap Oral vs Aboral:
  - upregulated only aboral (1000) only oral (0100)
  - upregulated both oral and aboral (1100)
  - downregulated only aboral (0010) only oral (0001)


# Set paths
If working from Laptop run:
```{r}
path<-"D:/Dropbox/PhD JCU/"
```

If working from JCU desktop run:
```{r}
#path<-"C:/Users/jc447193/Dropbox/PhD JCU/"
```


#1. Load results of all upsetplot intersections

```{r}
TopGO_0001<-read.csv("TopGO_Output/02_Upset_up_down/TopGO_0001.csv", header = T)
TopGO_0010<-read.csv("TopGO_Output/02_Upset_up_down/TopGO_0010.csv", header = T)

TopGO_0011<-read.csv("TopGO_Output/02_Upset_up_down/TopGO_0011.csv", header = T)
TopGO_0100<-read.csv("TopGO_Output/02_Upset_up_down/TopGO_0100.csv", header = T)

TopGO_1000<-read.csv("TopGO_Output/02_Upset_up_down/TopGO_1000.csv", header = T)
TopGO_1100<-read.csv("TopGO_Output/02_Upset_up_down/TopGO_1100.csv", header = T)
```


```{r}
# tbl_up_down_fig2 <-
#     list.files(path = paste0(path, "Chapter2_Gene expression/A.mil_oral_aboral_rnaseq/04_Functional_analysis/04_4_TopGO/TopGO_Output/Upset_up_down"),
#                pattern = "TopGO_[0-1][0-1][0-1][0-1].csv", 
#                full.names = T) %>% 
#     map_df(~read_csv(., col_types = cols(.default = "c"))) 
```


Load results from limma
```{r}
load(paste0(path,"Chapter2_Gene expression/A.mil_oral_aboral_rnaseq/03_Differential_gene_expression_analysis/Output/Complete_toptable/A.mil_oralaboral_Toptable_limma_trend_all_contrasts.RData"))
```

# 2. Upset_boxplot preparation

Defines function to prepare TopGO results into boxplot
```{r}
preparation_boxplot<-function(set, larvae_part, figure_part,limma_table){
prep<- set %>% mutate(larvae_part=paste0(larvae_part)) %>% mutate(figure_part=paste0(figure_part)) 
limma_table_2<-limma_table %>%  mutate(id_number = str_replace(amil_id, "amil.", "")) %>% mutate(gene_id=paste(id_number,".m1",sep=""))
prep<-prep %>% left_join(limma_table_2) %>% mutate(Regulation = ifelse(logFC>0, "Positive", "Negative")) 
}
```


```{r}
Fig_both_up_1100_oral<-preparation_boxplot(TopGO_1100, larvae_part = "Oral induced", figure_part = "both_up", table_oralEL_vs_oral)
Fig_both_up_1100_aboral<-preparation_boxplot(TopGO_1100, larvae_part = "Aboral induced", figure_part = "both_up", table_aboralEL_vs_aboral)
```


```{r}
Fig_up_0100_oral<-preparation_boxplot(TopGO_0100, larvae_part = "Oral induced", figure_part = "up", table_oralEL_vs_oral)
Fig_up_1000_aboral<-preparation_boxplot(TopGO_1000, larvae_part = "Aboral induced", figure_part = "up", table_aboralEL_vs_aboral)
```


```{r}
Fig_down_0001_oral<-preparation_boxplot(TopGO_0001, larvae_part = "Oral induced", figure_part = "down", table_oralEL_vs_oral)
Fig_down_0010_aboral<-preparation_boxplot(TopGO_0010, larvae_part = "Aboral induced", figure_part = "down", table_aboralEL_vs_aboral)
```


```{r}
Fig_both_down_0011_oral<-preparation_boxplot(TopGO_0011, larvae_part = "Oral induced", figure_part = "both_down", table_oralEL_vs_oral)
Fig_both_down_0011_aboral<-preparation_boxplot(TopGO_0011, larvae_part = "Aboral induced", figure_part = "both_down", table_aboralEL_vs_aboral)
```



combine results
```{r}
# Data_figure2_up_vs_down<- rbind(Fig_both_up_1100_oral,
#                                 Fig_both_up_1100_aboral,
#                                 Fig_up_0100_oral,
#                                 Fig_up_1000_aboral,
#                                 Fig_down_0001_oral,
#                                 Fig_down_0010_aboral,
#                                 Fig_both_down_0011_oral,
#                                 Fig_both_down_0011_aboral
#                                 
#                          )
# write.csv(Data_figure2_up_vs_down,"TopGO_Output/02_Upset_up_down/TopGO_figure/Data_figure2_up_vs_down_vision2.csv", row.names = FALSE)
# #Load manually annotated data
Data_figure2_up_vs_down<-read.csv("TopGO_Output/02_Upset_up_down/TopGO_figure/Data_figure2_up_vs_down_vision2_manually_anno.csv",na.string = "NA") 
```

Make columns factors
```{r}
Data_figure2_up_vs_down$larvae_part<-factor(Data_figure2_up_vs_down$larvae_part,levels = c("Aboral induced", "Oral induced"))
```

```{r}
levels(Data_figure2_up_vs_down$larvae_part)
```


```{r}
Data_figure2_up_vs_down$figure_part<-factor(Data_figure2_up_vs_down$figure_part,levels = c("up","both_up","down", "both_down"))
```


```{r}
levels(Data_figure2_up_vs_down$figure_part)
```



# 3. Create distance matrix and tree in function
Reference: https://ourcodingclub.github.io/tutorials/data-clustering/
From file: 12_Upset_boxplot_arranged_by_similarity_2021_05_14.Rmd
```{r}
create_distance_matrix<-function(figure_data,fig_part,larval_part, saving_path, file_name){
  # filter for data of specific figure part like both upregulated
  GOxgene_id<-figure_data %>% filter(figure_part==paste0(fig_part)) %>% filter(larvae_part==paste0(larval_part)) %>% 
    # select only columns of GO id and gene_id
    dplyr::select(GO.ID,gene_id) 
  
   #Checkpoint
  assign("check_point_tree",  GOxgene_id, envir = .GlobalEnv)
  
  # Use GO ID and gene ID to create matrix row and colums
  GO_sub <- unique(GOxgene_id$GO.ID)  # Making a vector with GO ids 
  gene_sub <- unique(GOxgene_id$gene_id)  # Making a vector with gene_ids
  
  # First we'll create an empty matrix with our sites in the rows and our genes in the columns. The       loop function will place a `1` on a given cell when the gene is present in an GO GO.ID and will       fill out the remaining cells with a `0`.

  commat <- matrix(0, length(GO_sub), length(gene_sub))
    for (i in 1:nrow(commat)){
    temp <- GOxgene_id[which(GOxgene_id$GO.ID== GO_sub[i]),]
    commat[i, which(gene_sub%in%temp$gene_id)] <- 1
    #print(i)
  }

  # Now let's name our rows and columns with the codes for the sites and the codes for the genes.
  rownames(commat) <- as.character(GOxgene_id$GO.ID[match(GO_sub, GOxgene_id$GO.ID)])
  colnames(commat) <- as.character(GOxgene_id$gene_id[match(gene_sub, GOxgene_id$gene_id)])

  # Calculate distances
  # dist <- recluster.dist(commat, dist="jaccard") #if you have sparse matrix (many zeroes) the best    #  choice is to use Jaccard index                                           
  #https://www.researchgate.net/post/How-to-choose-the-right-distance-method
  #dist <- recluster.dist(commat, dist="simpson")
  #dist <- recluster.dist(commat, dist="sorensen")
  #dist <- vegdist(commat, method="euclidean")
  
  
  clust <- recluster.cons(commat, tr = 1, p = 0.5, dist = "jaccard", method = "single")
  clust2  <- clust$cons  # Selecting the consensus tree 
  
  #Plot tree
  plot(clust2, direction = "downwards", cex = 0.5)

writeNexus(clust2, file=paste0(saving_path,file_name,".nex"))


}
```

```{r}
create_distance_matrix(figure_data=Data_figure2_up_vs_down,
                       fig_part="up", 
                       larval_part = "Aboral induced",
                       saving_path="TopGO_Output/02_Upset_up_down/TopGO_figure/", 
                       file_name="TopGO_figure_Aboral_up")
```


```{r}
create_distance_matrix(figure_data=Data_figure2_up_vs_down,
                       fig_part="up", 
                       larval_part = "Oral induced",
                       saving_path="TopGO_Output/02_Upset_up_down/TopGO_figure/", 
                       file_name="TopGO_figure_Oral_up")
```

```{r}
create_distance_matrix(figure_data=Data_figure2_up_vs_down,
                       fig_part="both_up", 
                       larval_part = "Aboral induced",
                       saving_path="TopGO_Output/02_Upset_up_down/TopGO_figure/", 
                       file_name="TopGO_figure_Aboral_both_up")
```


```{r}
create_distance_matrix(figure_data=Data_figure2_up_vs_down,
                       fig_part="both_up", 
                       larval_part = "Oral induced",
                       saving_path="TopGO_Output/02_Upset_up_down/TopGO_figure/", 
                       file_name="TopGO_figure_Oral_both_up")
```



```{r}
create_distance_matrix(figure_data=Data_figure2_up_vs_down,
                       fig_part="both_down", 
                       larval_part = "Aboral induced",
                       saving_path="TopGO_Output/02_Upset_up_down/TopGO_figure/", 
                       file_name="TopGO_figure_Aboral_both_down")
```


```{r}
create_distance_matrix(figure_data=Data_figure2_up_vs_down,
                       fig_part="both_down", 
                       larval_part = "Oral induced",
                       saving_path="TopGO_Output/02_Upset_up_down/TopGO_figure/", 
                       file_name="TopGO_figure_Oral_both_down")
```


```{r}
create_distance_matrix(figure_data=Data_figure2_up_vs_down,
                       fig_part="down", 
                       larval_part = "Aboral induced",
                       saving_path="TopGO_Output/02_Upset_up_down/TopGO_figure/", 
                       file_name="TopGO_figure_Aboral_down")
```


```{r}
create_distance_matrix(figure_data=Data_figure2_up_vs_down,
                       fig_part="down", 
                       larval_part = "Oral induced",
                       saving_path="TopGO_Output/02_Upset_up_down/TopGO_figure/", 
                       file_name="TopGO_figure_Oral_down")
```



```{r}
tree_up_aboral<-read.nexus(file="TopGO_Output/02_Upset_up_down/TopGO_figure/TopGO_figure_Aboral_up.nex", tree.names = NULL)
tree_up_oral<-read.nexus(file="TopGO_Output/02_Upset_up_down/TopGO_figure/TopGO_figure_Oral_up.nex", tree.names = NULL)


tree_both_up_aboral<-read.nexus(file="TopGO_Output/02_Upset_up_down/TopGO_figure/TopGO_figure_Aboral_both_up.nex", tree.names = NULL)
tree_both_up_oral<-read.nexus(file="TopGO_Output/02_Upset_up_down/TopGO_figure/TopGO_figure_Oral_both_up.nex", tree.names = NULL)

tree_down_aboral<-read.nexus(file="TopGO_Output/02_Upset_up_down/TopGO_figure/TopGO_figure_Aboral_down.nex", tree.names = NULL)
tree_down_oral<-read.nexus(file="TopGO_Output/02_Upset_up_down/TopGO_figure/TopGO_figure_Oral_down.nex", tree.names = NULL)

tree_both_down_aboral<-read.nexus(file="TopGO_Output/02_Upset_up_down/TopGO_figure/TopGO_figure_Aboral_both_down.nex", tree.names = NULL)
tree_both_down_oral<-read.nexus(file="TopGO_Output/02_Upset_up_down/TopGO_figure/TopGO_figure_Oral_both_down.nex", tree.names = NULL)

```


3 Manual creation of boxplot (optional) --> Useful to be able to write/understand function called create_similarity_arranged_boxplot
Plot tree without annotations
```{r}
p_tree_fig1<-ggtree(tree_up_aboral)+ layout_rectangular()  #+ ggplot2::xlim(0, 60) #xlim influences spache for brach lengths
#including xlim caused serious trouble
p_tree_fig1
```
Check this to know which terms to add in plot function

# 4. Define colours for manual annotation
```{r}
Data_figure2_up_vs_down %>%  pull(Manual_annotation) %>% unique()
```

```{r}
#I think the orger in this vector determines the order in the plot

col_pal<-c(
    "Signalling" ="#489FB5" ,         #light blue
    "Protein glycosylation" ="#0B3F5B",     #dark blue
    "Metabolism"="#6D224A", #purple
    "ROS" ="#A33814",       #red
    "Tissue remodelling" ="#D77316", #orange
    "Translation" ="#F5AF00",     #yellow
    "Transcription"="#3FB42D"        #light green

)

#489FB5 #light blue
#0B3F5B #dark blue
#6D224A OR #5D1D3F #purple
#A33814 OR #913212 #red
#D77316 OR #BA7611 OR #BB5B11 #orange
#F5AF00 #yellow
#3FB42D #light green
#107F22 #green 

```


# 5. Function Upsetplot arranged by similarity

I need different functions for the different plot parts since upregulated has different axis from downregulated and the chages in both larval parts require 2 boxplots due to different logFC values

Backup
```{r}
# create_similarity_arranged_boxplot_fig2<-function(enrich_GO_data, figu_part,nexus_tree, saving_path, file_name, col_pal){
#   
#   # Colour for manuall annotations
#   col_pal <- col_pal
#   
#   #Prepare data for tree annotations by filtering only one figure part e.g. up --> this should be already influenced by tree
#   data<-enrich_GO_data %>% filter(figure_part==paste0(figu_part)) %>% 
#     mutate(label=GO.ID) %>% 
#     mutate(Regulation = ifelse(logFC>0, "Positive", "Negative"))
# 
#   #Checkpoint
#   assign("data_test", data, envir = .GlobalEnv)
#   
#   #info = annotation for tree
#   info<- data %>% dplyr::select(label,Ontology,Term,GO.ID,weightFisher, Significant) %>% mutate(GO_Term=paste(GO.ID, Term, sep=" "))
#   #Checkpoint
#   assign("info_test", info, envir = .GlobalEnv)
# 
#   #Plot tree without annotations
#   tree<-ggtree(nexus_tree)+ layout_rectangular()#+ggplot2::xlim(c(0,-6,-6),c(200,6,6)) #xlim determines achis of all 3 pannels
#   #Checkpoint
#   assign("tree_test", tree, envir = .GlobalEnv)
# 
#   #Add annotations to tree
#   tree_anno <- tree %<+% info+
#     geom_tippoint(aes(shape=Ontology),colour = "grey30")+  # Change tip point to ontology shape
#     geom_tiplab(aes(label=GO_Term),offset = 0.1, size=2.5)+ xlim_tree(4.5)
#   #Checkpoint
#   assign("tree_anno_test",  tree_anno, envir = .GlobalEnv)
#   
#   #Prepare data to align to annotated tree
#   data_aboral_induced<- data %>% filter(larvae_part=="Aboral induced") %>%
#     dplyr::select(label,logFC,weightFisher, Significant,Manual_annotation_aboral) %>%
#     mutate(Aboral_EL_logFC=logFC) %>% 
#     mutate(Count_aboral=Significant) 
#    #Checkpoint
#   assign("data_aboral_test",  data_aboral_induced, envir = .GlobalEnv)
#   
#   
#   data_oral_induced<- data %>% filter(larvae_part=="Oral induced")  %>%
#     dplyr::select(label,logFC,weightFisher,Significant,Manual_annotation_oral) %>%
#     mutate(Oral_EL_logFC=logFC)%>% 
#     mutate(Count_oral=Significant)
# 
#   data_combined<-full_join(data_aboral_induced,data_oral_induced) %>% mutate(weightFisher_num=as.numeric(as.character(weightFisher))) 
#   #Checkpoint
#   assign("data_combined_Fig2",  data_combined, envir = .GlobalEnv)
#   
#    # Matrix for heatmap that annotate boxplots manually
#   #Manually_annotated_GO<-enrich_GO_data %>% dplyr::select(GO.ID, Manual_annotation) %>% distinct( .keep_all = T)  %>%column_to_rownames("GO.ID")
#  #  assign("Manually_annotated_GO",  Manually_annotated_GO, envir = .GlobalEnv)
#   
#    
#    #Function to create achsis different for oral and aboral
#   #https://coolbutuseless.github.io/2019/03/07/custom-axis-breaks-on-facetted-ggplot/
#   breaks_fun <- function(x) {
#   if (max(x) > 0) {
#       seq(0, 3, 1)
#     } else {
#       seq(-3 , 0, 1)
#      }
#   }
#   
#   
#   
#    #Function to create limits different for oral and aboral
#   limit_fun <- function(x) {
#   if (max(x) > 0) {
#       c(0, 3)
#     } else {
#        c((-3), 0)
#      }
#   }
# 
#   #Convert weightFisher to numeric
#   #data_combined$weightFisher<-as.numeric(as.character(data_combined$weightFisher))
#   
#     #Define colours for fill gradient for boxplots
#  # my_colours=c("#BEE0CC","#70C3D0","#419DC5","#316BA7","#223B89","#151E5E")
#  # my_colours=c("#151E5E","#223B89","#316BA7","#419DC5","#70C3D0","#BEE0CC")
# #my_colours=c("#BEE0CC","#70C3D0","#419DC5","#316BA7","#223B89","#151E5E")#blue
# my_colours=c("#808080","#9A9A9A","#B5B5B5","#D0D0D0","#EDEDED","#FFFFFF")#greys
# #my_colours2=c("green","#BEE0CC")
# 
# #define limits for legend
# limits_aboral<-enrich_GO_data %>% filter(figure_part==paste0(figu_part)) %>% filter(larvae_part=="Aboral induced")%>% filter(!is.na(Manual_annotation_aboral)) %>% pull(Manual_annotation_aboral) %>% unique()
# limits_oral<-enrich_GO_data %>% filter(figure_part==paste0(figu_part)) %>% filter(larvae_part=="Oral induced") %>% filter(!is.na(Manual_annotation_oral))%>% pull(Manual_annotation_oral) %>% unique()
# 
#   #Align data to annotated tree manual annotations
# tree_data_m<-  facet_plot(tree_anno ,panel = "N", geom = geom_tile,
#   aes(x = 1, fill=Manual_annotation_aboral), width = 2.5, data = data_combined) +# , colour = "black"
#     scale_fill_manual(values=col_pal, limits=limits_aboral,name="Manual annotations aboral",na.value="white")
# 
#  # add number of genees in each GO term onto manual annotation column
# tree_data_mn<- facet_plot(tree_data_m,panel = "N", data = data_combined, geom = geom_text,
#             aes(x = 1, label = Count_aboral))
#  
#             
# #New scale for next columns
# tree_data_mns <- tree_data_mn + new_scale_fill()
# 
# tree_data_mnsb<- tree_data_mns +   geom_facet(panel = "Aboral\ninduced", data = data_combined, geom =ggstance::geom_boxploth , #alternative geom_violinh
#               aes(x = Aboral_EL_logFC, group = label,fill=weightFisher_num), width = .6) +
#               geom_facet(panel = "Oral\ninduced", data = data_combined, geom = ggstance::geom_boxploth,
#               aes(x = Oral_EL_logFC, group = label,fill=weightFisher_num), width = .6)+
#               theme_tree2()+
#            # scale_x_continuous(breaks = breaks_fun,limits = limit_fun)+
#              scale_fill_gradientn(colours=my_colours,name="weightFisher")
# #New scale for next columns
# tree_data_mnsbs <- tree_data_mnsb + new_scale_fill()
# 
# tree_data_mnsbsm<-  facet_plot(tree_data_mnsbs ,panel = "N ", geom = geom_tile,
#    aes(x = 1, fill=Manual_annotation_oral),width = 2.5,  data = data_combined) +# , colour = "black"
#      scale_fill_manual(values=col_pal,limits=limits_oral,
#                          name="Manual annotations oral",na.value="white")
# tree_data_mnsbsm
# 
# tree_data_mnsbsmn<- facet_plot(tree_data_mnsbsm,panel = "N ", data = data_combined, geom = geom_text,
#              aes(x = 1, label = Count_oral))
# tree_data_mnsbsmn<-facet_labeller(tree_data_mnsbsmn, c(Tree = "Enriched Gene ontology terms"))
#  
# #change withs of columns
# gt = ggplot_gtable(ggplot_build(tree_data_mnsbsmn))
# assign("gt_check",  gt , envir = .GlobalEnv)
# 
# gt$widths[7] = 0.09*gt$widths[7] # in this case it was colmun 7 - reduce the width by a half
# gt$widths[13] = 0.09*gt$widths[13] # in this case it was colmun 7 - reduce the width by a half
# gt$widths[9] = 0.2*gt$widths[9] # in this case it was colmun 7 - reduce the width by a half
# gt$widths[11] = 0.2*gt$widths[11] # in this case it was colmun 7 - reduce the width by a half
# 
# ggsave(paste0(saving_path,file_name,".png"), gt,
#        width = 20,
#        height = 14,
#        units =  "cm")
# dev.off()
# 
# }
```

##Upregulated plot function

```{r}
create_similarity_arranged_boxplot_fig2_upregulated<-function(enrich_GO_data, figu_part,larval_part,nexus_tree, saving_path, file_name, col_pal, tree_label,plot_height){
  
  # Colour for manuall annotations
  col_pal <- col_pal
  
  #Prepare data for tree annotations by filtering only one figure part e.g. up --> this should be already influenced by tree
  data<-enrich_GO_data %>% filter(figure_part==paste0(figu_part)) %>%  filter(larvae_part==paste0(larval_part)) %>% 
    mutate(label=GO.ID)

  #Checkpoint
 # assign("data_test", data, envir = .GlobalEnv)
  
  #info = annotation for tree
  info<- data %>% dplyr::select(label,Ontology,Term,GO.ID,weightFisher, Significant, larvae_part, figure_part) %>% mutate(GO_Term=paste(GO.ID, Term, sep=" ")) 
  #Checkpoint
 # assign("info_test", info, envir = .GlobalEnv)

  #Plot tree without annotations
  tree<-ggtree(nexus_tree)+ layout_rectangular()#+ggplot2::xlim(c(0,-6,-6),c(200,6,6)) #xlim determines achis of all 3 pannels
  #Checkpoint
 # assign("tree_test", tree, envir = .GlobalEnv)

  #Add annotations to tree
  tree_anno <- tree %<+% info+
    geom_tippoint(aes(shape=Ontology),colour = "grey30")+  # Change tip point to ontology shape
    geom_tiplab(aes(label=GO_Term),offset = 0.1, size=3)+ xlim_tree(4.5)+
     scale_shape_discrete(name = "Gene ontology domains", labels = c("BP"="Biological Process", "CC"="Cellular Component", "MF"="Molecular Function"))
    
  #Checkpoint
  #assign("tree_anno_test",  tree_anno, envir = .GlobalEnv)
  
  #Prepare data to align to annotated tree
  data_induced<- data %>% filter(larvae_part==paste0(larval_part)) %>%
    dplyr::select(label,logFC,weightFisher, Significant,Manual_annotation) %>%
    mutate(Aboral_EL_logFC=logFC) %>% 
    mutate(Count_aboral=Significant) 
   #Checkpoint
  #assign("data_aboral_test",  data_induced, envir = .GlobalEnv)
  
  
  # data_oral_induced<- data %>% filter(larvae_part=="Oral induced")  %>%
  #   dplyr::select(label,logFC,weightFisher,Significant,Manual_annotation_oral) %>%
  #   mutate(Oral_EL_logFC=logFC)%>% 
  #   mutate(Count_oral=Significant)

  data_combined<-data_induced %>% mutate(weightFisher_num=as.numeric(as.character(weightFisher))) 
  #Checkpoint
  #assign("data_combined_Fig2",  data_combined, envir = .GlobalEnv)
  
   # Matrix for heatmap that annotate boxplots manually
  #Manually_annotated_GO<-enrich_GO_data %>% dplyr::select(GO.ID, Manual_annotation) %>% distinct( .keep_all = T)  %>%column_to_rownames("GO.ID")
 #  assign("Manually_annotated_GO",  Manually_annotated_GO, envir = .GlobalEnv)
  
   
   #Function to create achsis different for oral and aboral
  #https://coolbutuseless.github.io/2019/03/07/custom-axis-breaks-on-facetted-ggplot/
  breaks_fun <- function(x) {
  if (max(x) > 0) {
      seq(0, 5, 1)
    } else {
      seq(-5 , 0, 1)
     }
  }



   #Function to create limits different for oral and aboral
  limit_fun <- function(x) {
  if (max(x) > 0) {
      c(0, 5)
    } else {
       c((-5), 0)
     }
  }

  
  #Convert weightFisher to numeric
  #data_combined$weightFisher<-as.numeric(as.character(data_combined$weightFisher))
  
    #Define colours for fill gradient for boxplots
my_colours=c("#808080","#9A9A9A","#B5B5B5","#D0D0D0","#EDEDED","#FFFFFF")#greys


#define limits for legend
limits_aboral<-enrich_GO_data %>% filter(figure_part==paste0(figu_part)) %>% filter(larvae_part==paste0(larval_part))%>% filter(!is.na(Manual_annotation)) %>% pull(Manual_annotation) %>% unique()
#limits_oral<-enrich_GO_data %>% filter(figure_part==paste0(figu_part)) %>% filter(larvae_part=="Oral induced") %>% filter(!is.na(Manual_annotation_oral))%>% pull(Manual_annotation_oral) %>% unique()

  #Align data to annotated tree manual annotations
tree_data_m<-  facet_plot(tree_anno ,panel = "N", geom = geom_tile,
  aes(x = 2.5, fill=Manual_annotation), width = 5, data = data_combined) +# , colour = "black"
    scale_fill_manual(values=col_pal, 
                      #limits=limits_aboral,
                      name="Manual annotations",na.value="white")

 # add number of genees in each GO term onto manual annotation column
tree_data_mn<- facet_plot(tree_data_m,panel = "N", data = data_combined, geom = geom_text,
            aes(x = 2.5, label = Count_aboral))
 
            
#New scale for next columns
tree_data_mns <- tree_data_mn + new_scale_fill()

#Boxplot
# tree_data_mnsb<- tree_data_mns +   geom_facet(panel = "Aboral\ninduced", data = data_combined, geom =ggstance::geom_boxploth , #alternative geom_violinh
#               aes(x = Aboral_EL_logFC, group = label,fill=weightFisher_num), width = .6) +
#               geom_facet(panel = "Oral\ninduced", data = data_combined, geom = ggstance::geom_boxploth,
#               aes(x = Oral_EL_logFC, group = label,fill=weightFisher_num), width = .6)+
#               theme_tree2()+
#            # scale_x_continuous(breaks = breaks_fun,limits = limit_fun)+
#              scale_fill_gradientn(colours=my_colours,name="weightFisher")


tree_data_mnsb<- tree_data_mns +  
  
  #Barplot
              #geom_facet(panel = "GO numbers", data = data_combined, geom =geom_barh, #alternative geom_violinh
              #aes(x = Count_aboral,fill=weightFisher_num), stat='identity', width = .6) +
  #boxplot
              geom_facet(panel = "Effect size", data = data_combined, geom = ggstance::geom_boxploth,
              aes(x = Aboral_EL_logFC, group = label,fill=weightFisher_num), width = .6)+
              theme_tree2(legend.position="right")+
             scale_x_continuous(limits = c(0,5))+
              scale_fill_gradientn(colours=my_colours,name="Significance\nGO enrichment")+
  xlab("                                                                                                                    log2 foldchange")

#tree_data_mnsb<-tree_data_mnsb_l +theme(legend.position="none")





#New scale for next columns

# tree_data_mnsbs <- tree_data_mnsb + new_scale_fill()
# 
# tree_data_mnsbsm<-  facet_plot(tree_data_mnsbs ,panel = "N ", geom = geom_tile,
#    aes(x = 1, fill=Manual_annotation_oral),width = 2.5,  data = data_combined) +# , colour = "black"
#      scale_fill_manual(values=col_pal,limits=limits_oral,
#                          name="Manual annotations oral",na.value="white")
# tree_data_mnsbsm
# 
# tree_data_mnsbsmn<- facet_plot(tree_data_mnsbsm,panel = "N ", data = data_combined, geom = geom_text,
#              aes(x = 1, label = Count_oral))
tree_data_mnsb<-facet_labeller(tree_data_mnsb, c(Tree = paste0("Enriched GO terms in ",tree_label)))
 

tree_data_mnsb




# #change withs of columns see https://groups.google.com/g/bioc-ggtree/c/tZ0qkluBeGU/m/nkfWC9ixCwAJ
gt = ggplot_gtable(ggplot_build(tree_data_mnsb))
assign("gt_check",  gt , envir = .GlobalEnv)
gt$layout$l[grep('panel-1-2', gt$layout$name)] # you want to find the column specific to panel-2

gt$widths[7] = 0.1*gt$widths[7] # in this case it was colmun 7 - reduce the width by a half
# gt$widths[13] = 0.09*gt$widths[13] # in this case it was colmun 7 - reduce the width by a half
gt$widths[9] = 0.5*gt$widths[9] # in this case it was colmun 7 - reduce the width by a half
# gt$widths[11] = 0.2*gt$widths[11] # in this case it was colmun 7 - reduce the width by a half
# 

#Make height of the plot depending on the number of enriched GO terms
#height<-length(nexus_tree$tip.label)*1
#height

#convert gtable into ggplot
plot<-as.ggplot(gt)

#Save in enrironment for futher patchwork
assign(paste0(file_name),  plot, envir = .GlobalEnv)

#save ggplot figure
ggsave(paste0(saving_path,file_name,".png"), plot,
       width = 20,
       height = plot_height,
       units =  "cm",
       dpi=350)
dev.off()
}
```




```{r}
create_similarity_arranged_boxplot_fig2_upregulated(Data_figure2_up_vs_down, 
                                        figu_part="up",
                                        larval_part = "Aboral induced",
                                        nexus_tree=tree_up_aboral, 
                                        saving_path="TopGO_Output/02_Upset_up_down/TopGO_figure/Vision2/",      
                                        file_name="Boxplot_Fig2_Aboral_up2",
                                       col_pal=col_pal,
                                       tree_label="aboral",
                                       plot_height=9
                                      
                                        )
```


```{r}
create_similarity_arranged_boxplot_fig2_upregulated(Data_figure2_up_vs_down, 
                                        figu_part="up",
                                        larval_part = "Oral induced",
                                        nexus_tree=tree_up_oral, 
                                        saving_path="TopGO_Output/02_Upset_up_down/TopGO_figure/Vision2/", 
                                        file_name="Boxplot_Fig2_Oral_up2",
                                       col_pal=col_pal,
                                       tree_label="oral",
                                       plot_height=4
                                        
                                        )
```

##Downregulated plot function

```{r}
create_similarity_arranged_boxplot_fig2_downregulated<-function(enrich_GO_data, figu_part,larval_part,nexus_tree, saving_path, file_name, col_pal, tree_label,plot_height){
  
  # Colour for manuall annotations
  col_pal <- col_pal
  
  #Prepare data for tree annotations by filtering only one figure part e.g. up --> this should be already influenced by tree
  data<-enrich_GO_data %>% filter(figure_part==paste0(figu_part)) %>%  filter(larvae_part==paste0(larval_part)) %>% 
    mutate(label=GO.ID)#%>% 
  

  #Checkpoint
  #assign("data_test", data, envir = .GlobalEnv)
  
  #info = annotation for tree
  info<- data %>% dplyr::select(label,Ontology,Term,GO.ID,weightFisher, Significant, larvae_part, figure_part) %>% mutate(GO_Term=paste(GO.ID, Term, sep=" ")) 
  #Checkpoint
  #assign("info_test", info, envir = .GlobalEnv)

  #Plot tree without annotations
  tree<-ggtree(nexus_tree)+ layout_rectangular()#+ggplot2::xlim(c(0,-6,-6),c(200,6,6)) #xlim determines achis of all 3 pannels
  #Checkpoint
  #assign("tree_test", tree, envir = .GlobalEnv)

  #Add annotations to tree
  tree_anno <- tree %<+% info+
    geom_tippoint(aes(shape=Ontology),colour = "grey30")+  # Change tip point to ontology shape
    geom_tiplab(aes(label=GO_Term),offset = 0.1, size=3)+ xlim_tree(1.75)+
     scale_shape_discrete(name = "Gene ontology domains", labels = c("BP"="Biological Process", "CC"="Cellular Component", "MF"="Molecular Function"))
  #Checkpoint
  #assign("tree_anno_test",  tree_anno, envir = .GlobalEnv)
  
  #Prepare data to align to annotated tree
  data_induced<- data %>% filter(larvae_part==paste0(larval_part)) %>%
    dplyr::select(label,logFC,weightFisher, Significant,Manual_annotation) %>%
    mutate(Aboral_EL_logFC=logFC) %>% 
    mutate(Count_aboral=Significant) 
   #Checkpoint
 # assign("data_aboral_test",  data_induced, envir = .GlobalEnv)
  
  
  # data_oral_induced<- data %>% filter(larvae_part=="Oral induced")  %>%
  #   dplyr::select(label,logFC,weightFisher,Significant,Manual_annotation_oral) %>%
  #   mutate(Oral_EL_logFC=logFC)%>% 
  #   mutate(Count_oral=Significant)

  data_combined<-data_induced %>% mutate(weightFisher_num=as.numeric(as.character(weightFisher))) 
  #Checkpoint
 # assign("data_combined_Fig2",  data_combined, envir = .GlobalEnv)
  
   # Matrix for heatmap that annotate boxplots manually
  #Manually_annotated_GO<-enrich_GO_data %>% dplyr::select(GO.ID, Manual_annotation) %>% distinct( .keep_all = T)  %>%column_to_rownames("GO.ID")
 #  assign("Manually_annotated_GO",  Manually_annotated_GO, envir = .GlobalEnv)
  
   
   #Function to create achsis different for oral and aboral
  #https://coolbutuseless.github.io/2019/03/07/custom-axis-breaks-on-facetted-ggplot/
  # breaks_fun <- function(x) {
  # if (max(x) > 0) {
  #     seq(0, 5, 1)
  #   } else {
  #     seq(-5 , 0, 1)
  #    }
  # }



   #Function to create limits different for oral and aboral
  limit_fun <- function(x) {
  if (max(x) > 0) {
      c(0, 2)
    } else {
       c((-2), 0)
     }
  }

  
  #Convert weightFisher to numeric
  #data_combined$weightFisher<-as.numeric(as.character(data_combined$weightFisher))
  
    #Define colours for fill gradient for boxplots
my_colours=c("#808080","#9A9A9A","#B5B5B5","#D0D0D0","#EDEDED","#FFFFFF")#greys


#define limits for legend
limits_aboral<-enrich_GO_data %>% filter(figure_part==paste0(figu_part)) %>% filter(larvae_part==paste0(larval_part))%>% filter(!is.na(Manual_annotation)) %>% pull(Manual_annotation) %>% unique()
#limits_oral<-enrich_GO_data %>% filter(figure_part==paste0(figu_part)) %>% filter(larvae_part=="Oral induced") %>% filter(!is.na(Manual_annotation_oral))%>% pull(Manual_annotation_oral) %>% unique()

  #Align data to annotated tree manual annotations
tree_data_m<-  facet_plot(tree_anno ,panel = "N", geom = geom_tile,
  aes(x = 1, fill=Manual_annotation), width = 2, data = data_combined) +# , colour = "black"   # x is the center of the tile but the axis depends on the plot
    scale_fill_manual(values=col_pal, limits=limits_aboral,name="Manual annotations",na.value="white")

 # add number of genees in each GO term onto manual annotation column
tree_data_mn<- facet_plot(tree_data_m,panel = "N", data = data_combined, geom = geom_text,
            aes(x = 1, label = Count_aboral)) # x is defining the position of the text and depends on axis
 
            
#New scale for next columns
tree_data_mns <- tree_data_mn + new_scale_fill()

#Boxplot
# tree_data_mnsb<- tree_data_mns +   geom_facet(panel = "Aboral\ninduced", data = data_combined, geom =ggstance::geom_boxploth , #alternative geom_violinh
#               aes(x = Aboral_EL_logFC, group = label,fill=weightFisher_num), width = .6) +
#               geom_facet(panel = "Oral\ninduced", data = data_combined, geom = ggstance::geom_boxploth,
#               aes(x = Oral_EL_logFC, group = label,fill=weightFisher_num), width = .6)+
#               theme_tree2()+
#            # scale_x_continuous(breaks = breaks_fun,limits = limit_fun)+
#              scale_fill_gradientn(colours=my_colours,name="weightFisher")


tree_data_mnsb<- tree_data_mns +  
  
  #Barplot
              #geom_facet(panel = "GO numbers", data = data_combined, geom =geom_barh, #alternative geom_violinh
              #aes(x = Count_aboral,fill=weightFisher_num), stat='identity', width = .6) +
  #boxplot
              geom_facet(panel = "Effect size", data = data_combined, geom = ggstance::geom_boxploth,
              aes(x = Aboral_EL_logFC, group = label,fill=weightFisher_num), width = .6)+
              theme_tree2()+
             #scale_x_continuous(limits = c(-2,0))+
              scale_x_continuous(limits = limit_fun)+  #breaks = breaks_fun,
              scale_fill_gradientn(colours=my_colours,name="Significance\nGO enrichment")+
             theme(legend.position="none")+
      xlab("                                                                                                                     log2 foldchange")
#New scale for next columns

# tree_data_mnsbs <- tree_data_mnsb + new_scale_fill()
# 
# tree_data_mnsbsm<-  facet_plot(tree_data_mnsbs ,panel = "N ", geom = geom_tile,
#    aes(x = 1, fill=Manual_annotation_oral),width = 2.5,  data = data_combined) +# , colour = "black"
#      scale_fill_manual(values=col_pal,limits=limits_oral,
#                          name="Manual annotations oral",na.value="white")
# tree_data_mnsbsm
# 
# tree_data_mnsbsmn<- facet_plot(tree_data_mnsbsm,panel = "N ", data = data_combined, geom = geom_text,
#              aes(x = 1, label = Count_oral))

tree_data_mnsb<-facet_labeller(tree_data_mnsb, c(Tree = paste0("Enriched GO terms in ",tree_label)))


#tree_data_mnsb




# #change withs of columns see https://groups.google.com/g/bioc-ggtree/c/tZ0qkluBeGU/m/nkfWC9ixCwAJ
gt = ggplot_gtable(ggplot_build(tree_data_mnsb))
assign("gt_check",  gt , envir = .GlobalEnv)
gt$layout$l[grep('panel-1-2', gt$layout$name)] # you want to find the column specific to panel-2

gt$widths[7] = 0.1*gt$widths[7] # in this case it was colmun 7 - reduce the width by a half
# gt$widths[13] = 0.09*gt$widths[13] # in this case it was colmun 7 - reduce the width by a half
gt$widths[9] = 0.5*gt$widths[9] # in this case it was colmun 7 - reduce the width by a half
# gt$widths[11] = 0.2*gt$widths[11] # in this case it was colmun 7 - reduce the width by a half


#convert gtable into ggplot
plot<-as.ggplot(gt)

#Save in enrironment for futher patchwork
assign(paste0(file_name),  plot, envir = .GlobalEnv)

#save ggplot figure
ggsave(paste0(saving_path,file_name,".png"), plot,
       width = 20,
       height = plot_height,
       units =  "cm",
       dpi=350)
dev.off()

}
```

```{r}
create_similarity_arranged_boxplot_fig2_downregulated(Data_figure2_up_vs_down, 
                                        figu_part="down",
                                        larval_part = "Aboral induced",
                                        nexus_tree=tree_down_aboral, 
                                        saving_path="TopGO_Output/02_Upset_up_down/TopGO_figure/Vision2/",      
                                        file_name="Boxplot_Fig2_Aboral_down2",
                                        col_pal=col_pal,
                                        tree_label="aboral",
                                        plot_height=9
                                      
                                        )
```


```{r}
create_similarity_arranged_boxplot_fig2_downregulated(Data_figure2_up_vs_down, 
                                        figu_part="down",
                                        larval_part = "Oral induced",
                                        nexus_tree=tree_down_oral, 
                                        saving_path="TopGO_Output/02_Upset_up_down/TopGO_figure/Vision2/", 
                                        file_name="Boxplot_Fig2_Oral_down2",
                                       col_pal=col_pal,
                                        tree_label="oral",
                                       plot_height=9
                                        
                                        )
```


##Downregulated both plot function
```{r}
create_similarity_arranged_boxplot_fig2_downregulated_both<-function(enrich_GO_data, figu_part,larval_part,nexus_tree, saving_path, file_name, col_pal,  tree_label, plot_height){
  
  # Colour for manuall annotations
  col_pal <- col_pal
  
  #Prepare data for tree annotations by filtering only one figure part e.g. up --> this should be already influenced by tree
  data<-enrich_GO_data %>% filter(figure_part==paste0(figu_part)) %>%   
    mutate(label=GO.ID)#%>% 
   

  #Checkpoint
  #assign("data_test", data, envir = .GlobalEnv)
  
  #info = annotation for tree
  info<- data %>% dplyr::select(label,Ontology,Term,GO.ID,weightFisher, Significant, larvae_part, figure_part) %>% mutate(GO_Term=paste(GO.ID, Term, sep=" ")) 
  #Checkpoint
 # assign("info_test", info, envir = .GlobalEnv)

  #Plot tree without annotations
  tree<-ggtree(nexus_tree)+ layout_rectangular()#+ggplot2::xlim(c(0,-6,-6),c(200,6,6)) #xlim determines achis of all 3 pannels
  #Checkpoint
 # assign("tree_test", tree, envir = .GlobalEnv)

  #Add annotations to tree
  tree_anno <- tree %<+% info+
    geom_tippoint(aes(shape=Ontology),colour = "grey30")+  # Change tip point to ontology shape
    geom_tiplab(aes(label=GO_Term),offset = 0.1, size=3)+ xlim_tree(2)+
     scale_shape_discrete(name = "Gene ontology domains", labels = c("BP"="Biological Process", "CC"="Cellular Component", "MF"="Molecular Function"))
  #Checkpoint
  #assign("tree_anno_test",  tree_anno, envir = .GlobalEnv)
  
  #Prepare data to align to annotated tree
  data_induced<- data %>% #filter(larvae_part==paste0(larval_part)) %>%
    dplyr::select(label,logFC,weightFisher, Significant,Manual_annotation,larvae_part, figure_part) %>%
   # mutate(Aboral_EL_logFC=logFC) %>% 
    mutate(Count=Significant) 
   #Checkpoint
  #assign("data_aboral_test",  data_induced, envir = .GlobalEnv)
  
  
  # data_oral_induced<- data %>% filter(larvae_part=="Oral induced")  %>%
  #   dplyr::select(label,logFC,weightFisher,Significant,Manual_annotation_oral) %>%
  #   mutate(Oral_EL_logFC=logFC)%>% 
  #   mutate(Count_oral=Significant)

  data_combined<-data_induced %>% mutate(weightFisher_num=as.numeric(as.character(weightFisher))) 
  #Checkpoint
  assign("data_combined_Fig2",  data_combined, envir = .GlobalEnv)
  
  data_combined_aboral<-data_combined %>% filter(larvae_part=="Aboral induced")
  data_combined_oral<-data_combined %>% filter(larvae_part=="Oral induced")
  
   # Matrix for heatmap that annotate boxplots manually
  #Manually_annotated_GO<-enrich_GO_data %>% dplyr::select(GO.ID, Manual_annotation) %>% distinct( .keep_all = T)  %>%column_to_rownames("GO.ID")
 #  assign("Manually_annotated_GO",  Manually_annotated_GO, envir = .GlobalEnv)
  
   
   #Function to create achsis different for oral and aboral
  #https://coolbutuseless.github.io/2019/03/07/custom-axis-breaks-on-facetted-ggplot/
  # breaks_fun <- function(x) {
  # if (max(x) > 0) {
  #     seq(0, 5, 1)
  #   } else {
  #     seq(-5 , 0, 1)
  #    }
  # }



   #Function to create limits different for oral and aboral
  limit_fun <- function(x) {
  if (max(x) > 0) {
      c(0, 2)
    } else {
       c((-2), 0)
     }
  }

  
  #Convert weightFisher to numeric
  #data_combined$weightFisher<-as.numeric(as.character(data_combined$weightFisher))
  
    #Define colours for fill gradient for boxplots
my_colours=c("#808080","#9A9A9A","#B5B5B5","#D0D0D0","#EDEDED","#FFFFFF")#greys


#define limits for legend
#limits_aboral<-enrich_GO_data %>% filter(figure_part==paste0(figu_part)) %>% filter(larvae_part==paste0(larval_part))%>% filter(!is.na(Manual_annotation)) %>% pull(Manual_annotation) %>% unique()
#limits_oral<-enrich_GO_data %>% filter(figure_part==paste0(figu_part)) %>% filter(larvae_part=="Oral induced") %>% filter(!is.na(Manual_annotation_oral))%>% pull(Manual_annotation_oral) %>% unique()

  #Align data to annotated tree manual annotations
tree_data_m<-  facet_plot(tree_anno ,panel = "N", geom = geom_tile,
  aes(x = 1, fill=Manual_annotation), width = 2, data = data_combined) +# , colour = "black"   # x is the center of the tile but the axis depends on the plot
    scale_fill_manual(values=col_pal, name="Manual annotations",na.value="white") #limits=limits_aboral,

 # add number of genees in each GO term onto manual annotation column
tree_data_mn<- facet_plot(tree_data_m,panel = "N", data = data_combined, geom = geom_text,
            aes(x = 1, label = Count)) # x is defining the position of the text and depends on axis
 
            
#New scale for next columns
tree_data_mns <- tree_data_mn + new_scale_fill()

#Boxplot
# tree_data_mnsb<- tree_data_mns +   geom_facet(panel = "Aboral\ninduced", data = data_combined, geom =ggstance::geom_boxploth , #alternative geom_violinh
#               aes(x = Aboral_EL_logFC, group = label,fill=weightFisher_num), width = .6) +
#               geom_facet(panel = "Oral\ninduced", data = data_combined, geom = ggstance::geom_boxploth,
#               aes(x = Oral_EL_logFC, group = label,fill=weightFisher_num), width = .6)+
#               theme_tree2()+
#            # scale_x_continuous(breaks = breaks_fun,limits = limit_fun)+
#              scale_fill_gradientn(colours=my_colours,name="weightFisher")


tree_data_mnsb<- tree_data_mns +  
  
  #Barplot
              #geom_facet(panel = "GO numbers", data = data_combined, geom =geom_barh, #alternative geom_violinh
              #aes(x = Count_aboral,fill=weightFisher_num), stat='identity', width = .6) +
  # #boxplot
              geom_facet(panel = "Effect size", data = data_combined_aboral, geom = ggstance::geom_boxploth,
              aes(x = logFC, group = label,fill="aboral"), width = .6, alpha=0.5)+
             # theme_tree2()+
             #scale_x_continuous(limits = c(-2,0))+
#  scale_x_continuous(limits = limit_fun)+  #breaks = breaks_fun,
            #  scale_fill_gradientn(colours=my_colours,name="weightFisher")+
  
    geom_facet(panel = "Effect size", data = data_combined_oral, geom = ggstance::geom_boxploth,
              aes(x = logFC, group = label,fill="oral"), width = .6, alpha=0.5)+
           theme_tree2()+
             scale_x_continuous(limits = c(-2,0))+
            scale_x_continuous(limits = limit_fun)+
            scale_fill_discrete(name="Larval part")+
             theme(legend.position="none")+
      xlab("                                                                                                                           log2 foldchange")

      #+  #breaks = breaks_fun,
             # scale_fill_gradientn(colours=my_colours,name="weightFisher")

  #points


  #             geom_facet(panel = "SNP", data =  data_combined_oral, geom = geom_point, 
  #              mapping=aes(x = logFC, colour = "red"), shape = '|') +
  # geom_facet(panel = "SNP", data =  data_combined_aboral, geom = geom_point, 
  #              mapping=aes(x = logFC, colour = "blue"), shape = '|')



#New scale for next columns

# tree_data_mnsbs <- tree_data_mnsb + new_scale_fill()
# 
# tree_data_mnsbsm<-  facet_plot(tree_data_mnsbs ,panel = "N ", geom = geom_tile,
#    aes(x = 1, fill=Manual_annotation_oral),width = 2.5,  data = data_combined) +# , colour = "black"
#      scale_fill_manual(values=col_pal,limits=limits_oral,
#                          name="Manual annotations oral",na.value="white")
# tree_data_mnsbsm
# 
# tree_data_mnsbsmn<- facet_plot(tree_data_mnsbsm,panel = "N ", data = data_combined, geom = geom_text,
#              aes(x = 1, label = Count_oral))

tree_data_mnsb<-facet_labeller(tree_data_mnsb, c(Tree = paste0("Enriched GO terms in ",tree_label))) 

tree_data_mnsb




# #change withs of columns see https://groups.google.com/g/bioc-ggtree/c/tZ0qkluBeGU/m/nkfWC9ixCwAJ
gt = ggplot_gtable(ggplot_build(tree_data_mnsb))
assign("gt_check",  gt , envir = .GlobalEnv)
gt$layout$l[grep('panel-1-2', gt$layout$name)] # you want to find the column specific to panel-2

gt$widths[7] = 0.1*gt$widths[7] # in this case it was colmun 7 - reduce the width by a half
# gt$widths[13] = 0.09*gt$widths[13] # in this case it was colmun 7 - reduce the width by a half
gt$widths[9] = 0.5*gt$widths[9] # in this case it was colmun 7 - reduce the width by a half
# gt$widths[11] = 0.2*gt$widths[11] # in this case it was colmun 7 - reduce the width by a half

#convert gtable into ggplot
plot<-as.ggplot(gt)

#Save in enrironment for futher patchwork
assign(paste0(file_name),  plot, envir = .GlobalEnv)

#save ggplot figure
ggsave(paste0(saving_path,file_name,".png"), plot,
       width = 20,
       height = plot_height,
       units =  "cm",
       dpi=350)
dev.off()

}
```

```{r}
create_similarity_arranged_boxplot_fig2_downregulated_both(Data_figure2_up_vs_down, 
                                        figu_part="both_down",
                                        larval_part = "Oral and Aboral induced",
                                        nexus_tree=tree_both_down_oral, 
                                        saving_path="TopGO_Output/02_Upset_up_down/TopGO_figure/Vision2/", 
                                        file_name="Boxplot_Fig2_Both_down2",
                                        col_pal=col_pal,
                                         tree_label="both larval parts",
                                       plot_height=9
                                        
                                        )
```


##Upregulated both plot function
```{r}
create_similarity_arranged_boxplot_fig2_upregulated_both<-function(enrich_GO_data, figu_part,larval_part,nexus_tree, saving_path, file_name, col_pal, tree_label,plot_height){
  
  # Colour for manuall annotations
  col_pal <- col_pal
  
  #Prepare data for tree annotations by filtering only one figure part e.g. up --> this should be already influenced by tree
  data<-enrich_GO_data %>% filter(figure_part==paste0(figu_part)) %>% 
    mutate(label=GO.ID)#%>% 


  #Checkpoint
  #assign("data_test", data, envir = .GlobalEnv)
  
  #info = annotation for tree
  info<- data %>% dplyr::select(label,Ontology,Term,GO.ID,weightFisher, Significant, larvae_part, figure_part) %>% mutate(GO_Term=paste(GO.ID, Term, sep=" ")) 
  #Checkpoint
  #assign("info_test", info, envir = .GlobalEnv)

  #Plot tree without annotations
  tree<-ggtree(nexus_tree)+ layout_rectangular()#+ggplot2::xlim(c(0,-6,-6),c(200,6,6)) #xlim determines achis of all 3 pannels
  #Checkpoint
  #assign("tree_test", tree, envir = .GlobalEnv)

  #Add annotations to tree
  tree_anno <- tree %<+% info+
    geom_tippoint(aes(shape=Ontology),colour = "grey30")+  # Change tip point to ontology shape
    geom_tiplab(aes(label=GO_Term),offset = 0.1, size=3)+ xlim_tree(4.5)+
     scale_shape_discrete(name = "Gene ontology domains", labels = c("BP"="Biological Process", "CC"="Cellular Component", "MF"="Molecular Function"))
  #Checkpoint
  #assign("tree_anno_test",  tree_anno, envir = .GlobalEnv)
  
  #Prepare data to align to annotated tree
  data_induced<- data %>% 
    dplyr::select(label,logFC,weightFisher, Significant,Manual_annotation,larvae_part, figure_part) %>%
   # mutate(Aboral_EL_logFC=logFC) %>% 
    mutate(Count=Significant) 
   #Checkpoint
  #assign("data_aboral_test",  data_induced, envir = .GlobalEnv)
  
  
  # data_oral_induced<- data %>% filter(larvae_part=="Oral induced")  %>%
  #   dplyr::select(label,logFC,weightFisher,Significant,Manual_annotation_oral) %>%
  #   mutate(Oral_EL_logFC=logFC)%>% 
  #   mutate(Count_oral=Significant)

  data_combined<-data_induced %>% mutate(weightFisher_num=as.numeric(as.character(weightFisher))) 
  #Checkpoint
  #assign("data_combined_Fig2",  data_combined, envir = .GlobalEnv)
  
  data_combined_aboral<-data_combined %>% filter(larvae_part=="Aboral induced")
  data_combined_oral<-data_combined %>% filter(larvae_part=="Oral induced")
  
   # Matrix for heatmap that annotate boxplots manually
  #Manually_annotated_GO<-enrich_GO_data %>% dplyr::select(GO.ID, Manual_annotation) %>% distinct( .keep_all = T)  %>%column_to_rownames("GO.ID")
 #  assign("Manually_annotated_GO",  Manually_annotated_GO, envir = .GlobalEnv)
  
   
   #Function to create achsis different for oral and aboral
  #https://coolbutuseless.github.io/2019/03/07/custom-axis-breaks-on-facetted-ggplot/
  # breaks_fun <- function(x) {
  # if (max(x) > 0) {
  #     seq(0, 5, 1)
  #   } else {
  #     seq(-5 , 0, 1)
  #    }
  # }



   #Function to create limits different for oral and aboral
  limit_fun <- function(x) {
  if (max(x) > 0) {
      c(0, 5)
    } else {
       c((-5), 0)
     }
  }

  
  #Convert weightFisher to numeric
  #data_combined$weightFisher<-as.numeric(as.character(data_combined$weightFisher))
  
    #Define colours for fill gradient for boxplots
my_colours=c("#808080","#9A9A9A","#B5B5B5","#D0D0D0","#EDEDED","#FFFFFF")#greys


#define limits for legend
#limits_aboral<-enrich_GO_data %>% filter(figure_part==paste0(figu_part)) %>% filter(larvae_part==paste0(larval_part))%>% filter(!is.na(Manual_annotation)) %>% pull(Manual_annotation) %>% unique()
#limits_oral<-enrich_GO_data %>% filter(figure_part==paste0(figu_part)) %>% filter(larvae_part=="Oral induced") %>% filter(!is.na(Manual_annotation_oral))%>% pull(Manual_annotation_oral) %>% unique()

  #Align data to annotated tree manual annotations
tree_data_m<-  facet_plot(tree_anno ,panel = "N", geom = geom_tile,
  aes(x = 2.5, fill=Manual_annotation), width = 5, data = data_combined) +# , colour = "black"   # x is the center of the tile but the axis depends on the plot
    scale_fill_manual(values=col_pal, 
                     # limits=limits_aboral,
                      name="Manual annotations",na.value="white")




 # add number of genees in each GO term onto manual annotation column
tree_data_mn<- facet_plot(tree_data_m,panel = "N", data = data_combined, geom = geom_text,
            aes(x = 2.5, label = Count)) # x is defining the position of the text and depends on axis
 
            
#New scale for next columns
tree_data_mns <- tree_data_mn + new_scale_fill()

#Boxplot
# tree_data_mnsb<- tree_data_mns +   geom_facet(panel = "Aboral\ninduced", data = data_combined, geom =ggstance::geom_boxploth , #alternative geom_violinh
#               aes(x = Aboral_EL_logFC, group = label,fill=weightFisher_num), width = .6) +
#               geom_facet(panel = "Oral\ninduced", data = data_combined, geom = ggstance::geom_boxploth,
#               aes(x = Oral_EL_logFC, group = label,fill=weightFisher_num), width = .6)+
#               theme_tree2()+
#            # scale_x_continuous(breaks = breaks_fun,limits = limit_fun)+
#              scale_fill_gradientn(colours=my_colours,name="weightFisher")


tree_data_mnsb<- tree_data_mns +  
  
  #Barplot
              #geom_facet(panel = "GO numbers", data = data_combined, geom =geom_barh, #alternative geom_violinh
              #aes(x = Count_aboral,fill=weightFisher_num), stat='identity', width = .6) +
  # #boxplot
  
             geom_facet(panel = "Effect size", data = data_combined_oral, geom = ggstance::geom_boxploth,
              aes(x = logFC, group = label,fill="Oral induced"), width = .6, alpha=0.5)+
  
              geom_facet(panel = "Effect size", data = data_combined_aboral, geom = ggstance::geom_boxploth,
              aes(x = logFC, group = label,fill="Aboral induced"), width = .6, alpha=0.5)+
              theme_tree2()+
              scale_x_continuous(limits = limit_fun)+  #breaks = breaks_fun,
              theme(legend.position="right")+
     xlab("                                                                                                                    log2 foldchange")
  
        
             #scale_x_continuous(limits = c(-2,0))+
            #  scale_x_continuous(limits = limit_fun)+  #breaks = breaks_fun,
            #  scale_fill_gradientn(colours=my_colours,name="weightFisher")+
  
     
  
#tree_data_mnsb<-tree_data_mnsb_l + theme(legend.position="none")
           
#Save legend for combined plots
#legend <- cowplot::get_legend(tree_data_mnsb_l)
#assign("legend",  legend, envir = .GlobalEnv)


#New scale for next columns

# tree_data_mnsbs <- tree_data_mnsb + new_scale_fill()
# 
# tree_data_mnsbsm<-  facet_plot(tree_data_mnsbs ,panel = "N ", geom = geom_tile,
#    aes(x = 1, fill=Manual_annotation_oral),width = 2.5,  data = data_combined) +# , colour = "black"
#      scale_fill_manual(values=col_pal,limits=limits_oral,
#                          name="Manual annotations oral",na.value="white")
# tree_data_mnsbsm
# 
# tree_data_mnsbsmn<- facet_plot(tree_data_mnsbsm,panel = "N ", data = data_combined, geom = geom_text,
#              aes(x = 1, label = Count_oral))

tree_data_mnsb<-facet_labeller(tree_data_mnsb, c(Tree = paste0("Enriched GO terms in ",tree_label))) 

tree_data_mnsb




# #change withs of columns see https://groups.google.com/g/bioc-ggtree/c/tZ0qkluBeGU/m/nkfWC9ixCwAJ
gt = ggplot_gtable(ggplot_build(tree_data_mnsb))
assign("gt_check",  gt , envir = .GlobalEnv)
gt$layout$l[grep('panel-1-2', gt$layout$name)] # you want to find the column specific to panel-2

gt$widths[7] = 0.1*gt$widths[7] # in this case it was colmun 7 - reduce the width by a half
# gt$widths[13] = 0.09*gt$widths[13] # in this case it was colmun 7 - reduce the width by a half
gt$widths[9] = 0.5*gt$widths[9] # in this case it was colmun 7 - reduce the width by a half
# gt$widths[11] = 0.2*gt$widths[11] # in this case it was colmun 7 - reduce the width by a half

#convert gtable into ggplot
plot<-as.ggplot(gt)

#Save in enrironment for futher patchwork
assign(paste0(file_name),  plot, envir = .GlobalEnv)

#save ggplot figure
ggsave(paste0(saving_path,file_name,".png"), plot,
       width = 20,
       height = plot_height,
       units =  "cm",
       dpi=350)
dev.off()


}
```

```{r}
create_similarity_arranged_boxplot_fig2_upregulated_both(Data_figure2_up_vs_down, 
                                        figu_part="both_up",
                                        larval_part = "Oral and Aboral induced",
                                        nexus_tree=tree_both_up_oral, 
                                        saving_path="TopGO_Output/02_Upset_up_down/TopGO_figure/Vision2/", 
                                        file_name="Boxplot_Fig2_Both_up2",
                                       col_pal=col_pal,
                                        tree_label="both larval parts",
                                       plot_height=9
                                        
                                        )
```
check out if i need limits
--> limits make only annotation that occur not all

#6. Combine plots using patchwork package
```{r}

figure_upregulated<- Boxplot_Fig2_Aboral_up2/ Boxplot_Fig2_Both_up2 / Boxplot_Fig2_Oral_up2 & theme(legend.position = "right")
figure_upregulated_anno<-figure_upregulated + plot_annotation(
                                                    #title = 'The surprising truth about mtcars',
                                                    #subtitle = 'These 3 plots will reveal yet-untold secrets about our beloved data-set',
                                                    #caption = 'Disclaimer: None of these plots are insightful',
                                tag_levels = 'A'
                                )+
  plot_layout(guides = "collect",
              heights =  c(1, 1,0.4))


ggsave(paste0("TopGO_Output/02_Upset_up_down/TopGO_figure/Vision2/","Combined_figure_upregulated3_legend",".png"), figure_upregulated_anno,
       width = 20,
       height = 27,
       units =  "cm",
       dpi=300)

```

```{r}

figure_downregulated<- Boxplot_Fig2_Aboral_down2/ Boxplot_Fig2_Both_down2 / Boxplot_Fig2_Oral_down2& theme(legend.position = "bottom")
figure_downregulated_anno<-figure_downregulated + plot_annotation(
                                                    #title = 'The surprising truth about mtcars',
                                                    #subtitle = 'These 3 plots will reveal yet-untold secrets about our beloved data-set',
                                                    #caption = 'Disclaimer: None of these plots are insightful',
                                tag_levels = 'A'
)+
  plot_layout(guides = "collect",
              heights =  c(1.1, 0.5,0.7))



ggsave(paste0("TopGO_Output/02_Upset_up_down/TopGO_figure/Vision2/","Combined_figure_downregulated_no_legend",".png"), figure_downregulated_anno,
       width = 20,
       height = 27,
       units =  "cm",
       dpi=300)

```

Next step:

-define hight of figure based on number of go terms?
-only one legend
-add larval figure
-add axis label below



Problem: I need 2 sets of boxplots for both up and both down. However the ggtree annotation with data only allows one boxplot per tree tip label
last try is to have a tree with GO.ID_larval part to have 2 tips per go term, so I can add the according boxplot. This needs a new tree with merged labels of goid and larvae part

or see https://yulab-smu.top/treedata-book/chapter7.html with points for each logFC coloured in oral and aboral
--> you can layer the geoms if the panel has the same name






run oral vs aboral with this function